File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/command/solution/report_position_command.cpp
// report_position_command.cpp

#include "driver.h"
#include "report_position_command.h"

report_position_command::report_position_command(driver * a_driver)
: simple_driver_command(a_driver)
{
}

void report_position_command::execute()
{
	get_driver()->report_position();
}

void report_position_command::undo()
{
	get_driver()->report_position();
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/command/solution/ReportPositionCommand.java
// ReportPositionCommand.java

public class ReportPositionCommand extends SimpleDriverCommand
{
	public ReportPositionCommand(Driver aDriver)
	{
		super(aDriver);
	}

	public void execute()
	{
		getDriver().reportPosition();
	}

	public void undo()
	{
		getDriver().reportPosition();
	}
}
